Packs/ThalesCipherTrustManager/Integrations/ThalesCipherTrustManager/ThalesCipherTrustManager.py

```diff
@@ -0,0 +1,919 @@
+    DURATION = 'duration'
+    NOT_AFTER = 'not_after'
+    NOT_BEFORE = 'not_before'
+    PARENT_ID = 'parent_id'
+    CA_ID = 'ca_id'
+    CSR = 'csr'
+    CSR_ENTRY_ID = 'csr_entry_id'
+    PURPOSE = 'purpose'
+    ID = 'id'
+    CERT_ID = 'cert_id'
+    REASON = 'reason'
+    PARENT = 'parent'
+    EXTERNAL_CA_ID = 'external_ca_id'
+    SERIAL_NUMBER = 'serial_number'
+    EXTERNAL_CERT_ID = 'external_cert_id'
+
+
+'''CLIENT CLASS'''
+
+
+class CipherTrustClient(BaseClient):
+    """ A client class to interact with the Thales CipherTrust API """
+
+    def __init__(self, username: str, password: str, server_url: str, proxy: bool, verify: bool, api_version='v1'):
+        base_url = urljoin(server_url, f'api/{api_version}')
+        super().__init__(base_url=base_url, proxy=proxy, verify=verify)
+        res = self.create_auth_token(username, password)
+        self._headers = {'Authorization': f'Bearer {res.get("jwt")}', 'accept': 'application/json'}
+
+    def create_auth_token(self, username: str, password: str):
+        return self._http_request(
+            method='POST',
+            url_suffix=AUTHENTICATION_URL_SUFFIX,
+            json_data={
+                'grant_type': 'password',
+                'username': username,
+                'password': password
+            },
+        )
+
+    def get_group_list(self, params: dict) -> dict[str, Any]:
+        return self._http_request(
+            method='GET',
+            url_suffix=USER_MANAGEMENT_GROUPS_URL_SUFFIX,
+            params=params,
+        )
+
+    def create_group(self, request_data: dict) -> dict[str, Any]:
+        return self._http_request(
+            method='POST',
+            url_suffix=USER_MANAGEMENT_GROUPS_URL_SUFFIX,
+            json_data=request_data,
+        )
+
+    def delete_group(self, group_name: str, request_data: dict) -> dict[str, Any]:
+        return self._http_request(
+            method='DELETE',
+            url_suffix=urljoin(USER_MANAGEMENT_GROUPS_URL_SUFFIX, quote(group_name)),
+            json_data=request_data,
+            return_empty_response=True,
+        )
+
+    def update_group(self, group_name: str, request_data: dict) -> dict[str, Any]:
+        return self._http_request(
+            method='PATCH',
+            url_suffix=urljoin(USER_MANAGEMENT_GROUPS_URL_SUFFIX, quote(group_name)),
+            json_data=request_data,
+        )
+
+    def add_user_to_group(self, group_name: str, user_id: str) -> dict[str, Any]:
+        return self._http_request(
+            method='POST',
+            url_suffix=f'{urljoin(USER_MANAGEMENT_GROUPS_URL_SUFFIX, quote(group_name))}/users/{user_id}',
+        )
+
+    def remove_user_from_group(self, group_name: str, user_id: str) -> dict[str, Any]:
+        return self._http_request(
+            method='DELETE',
+            url_suffix=f'{urljoin(USER_MANAGEMENT_GROUPS_URL_SUFFIX, quote(group_name))}/users/{user_id}',
+            return_empty_response=True,
```

Comment:
raised exception otherwise since the api returns an empty response. 
see _handle_sucsses method that is called by the _http_request: 
![image](https://github.com/demisto/content/assets/94686128/df523b14-f8e7-46be-b16d-66b273847673)
![image](https://github.com/demisto/content/assets/94686128/aecb9b82-2128-4581-a172-54d23dce4e25)
![image](https://github.com/demisto/content/assets/94686128/bfb6c922-f0a2-46fc-a320-0641d389b165)